--- a/py/private/py_venv/py_venv_exec.bzl
+++ b/py/private/py_venv/py_venv_exec.bzl
@@ -10,6 +10,7 @@
 load("@hermetic_launcher//launcher:lib.bzl", "launcher")
 load("@rules_python//python:defs.bzl", "PyInfo")
 load("//py/private:py_semantics.bzl", _py_semantics = "semantics")
+load("//py/private/toolchain:types.bzl", "PY_TOOLCHAIN")
 load(":types.bzl", "VirtualenvInfo", "venv_root")
 
 # Identifiers the launcher always sets to the analysing rule's contextual
@@ -246,13 +247,13 @@
     implementation = _py_venv_exec_impl,
     attrs = _attrs,
     executable = True,
-    toolchains = [launcher.finalizer_toolchain_type, launcher.template_toolchain_type],
+    toolchains = [launcher.finalizer_toolchain_type, launcher.template_toolchain_type, PY_TOOLCHAIN],
 )
 
 py_venv_exec_test = rule(
     doc = "Test variant of `py_venv_exec`. Most users should use the [py_test macro](#py_test) instead of loading this directly.",
     implementation = _py_venv_exec_impl,
     attrs = _attrs | _test_attrs,
     test = True,
-    toolchains = [launcher.finalizer_toolchain_type, launcher.template_toolchain_type],
+    toolchains = [launcher.finalizer_toolchain_type, launcher.template_toolchain_type, PY_TOOLCHAIN],
 )
